Add information on python metrics to the programming guide (#32464)
apache/beam: website/www/site/content/en/documentation/programming-guide.md

be used for custom filters.
{{< /paragraph >}}

{{< paragraph class="language-py">}}

> `PipelineResult` has a method `metrics()` which returns a `MetricResults` object that allows

Reviewer: ```suggestion
`PipelineResult` has a `metrics` method that returns a `MetricResults` object. The `MetricResults` object  lets you
```

> accessing metrics. The main method available in a `MetricResults` object, `query()`, allows

Reviewer: ```suggestion
access metrics. The main method available in the `MetricResults` object, `query`, lets you
```

> for querying all metrics matching a given filter. It takes in a `MetricsFilter` object that can

Reviewer: ```suggestion
query all metrics that match a given filter. The `query` method takes in a `MetricsFilter` object that you can
```

> be constructed to filter by several different criteria. Querying a `MetricResults` object returns

Reviewer: ```suggestion
use to filter by several different criteria. Querying a `MetricResults` object returns
```

> a dictionary of lists of `MetricResult` objects, with the dictionary organizing them by type

Reviewer: ```suggestion
a dictionary of lists of `MetricResult` objects, with the dictionary organizing them by type,
```
